### PR TITLE
store: fix misc label_value issues related to external labels

### PR DESCRIPTION
**Pull request by MichaHoffmann**
If the requested label was an external label, we would ignore matchers in tsdb and prometheus store.

### CODE REVIEW

File: pkg/store/prometheus.go [hunk 1]
```diff
@@ -709,6 +709,11 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 
 	// First check for matching external label which has priority.
 	if l := extLset.Get(r.Label); l != "" {
+		for _, m := range matchers {
+			if !m.Matches(l) {
+				return &storepb.LabelValuesResponse{Values: nil}, nil
```
Comment (GiedriusS) [inline pkg/store/prometheus.go:714]: I don't think we need to explicitly set `Values: nil` because that's the zero value for that field, no?

**Review by GiedriusS — APPROVED** [review]

File: pkg/store/prometheus.go (hunk 1 above)
Comment (MichaHoffmann) [reply pkg/store/prometheus.go:714]: Yeah thats true!

Comment (MichaHoffmann) [reply pkg/store/prometheus.go]: Done ( sorry for the delay ); i also changed the other instances of this to be uniform.

**Review by yeya24 — APPROVED** [review]
Thanks!

**Review by bwplotka — APPROVED** [review]
Nice! Thanks 👍🏽

**Comment by sepich** [comment]
Could you please port this to v0.33 also?
This is very important fix, but not even in the changelog

**Comment by MichaHoffmann** [comment]
> Could you please port this to v0.33 also? This is very important fix, but not even in the changelog

Oh, yeah sorry, it seemed just like a small fix at the time! Let me cherry pick it a bit later!

**Comment by MichaHoffmann** [comment]
> Could you please port this to v0.33 also? This is very important fix, but not even in the changelog

This is included in 0.34.0!

**Pull request merged**